### ISSUE TITLE
Bitfinex trade symbol non-normalized/formatted correctly

### ISSUE THREAD

**Issue by bobbytaxbit** [ccxt/ccxt#6632]
- OS: Ubuntu
- Programming Language version: Python 3.8
- CCXT version: 1.24.7
- Exchange: Bitfinex2
- Method: fetch_my_trades

Making a request to fetch_my_trades returns data as expected, but it appears that in _some_ cases the 'symbol' field is not returning a properly formatted trading pair. According to the manual:

```
symbol. An uppercase string code representation of a particular trading pair or instrument. This is usually written as BaseCurrency/QuoteCurrency with a slash as in BTC/USD, LTC/CNY or ETH/EUR, etc. Symbols are used to reference markets within the ccxt library (explained below).
```
I am getting back the following symbols in the Trade Entity (in addition to the one's I would expect) on a few trades.

```
'tBCHBTC', 'tBCHETH', 'tBCHUSD'
```

These don't have the slash between the currencies and BTC is proceeded with the lowercase `t` which doesn't appear to be normalized like I'm used to with other CCXT integrations.

Also, if I look at the `markets` available, I don't see see any of those listed for bitfinex2

```
['BTC/USD', 'LTC/USD', 'LTC/BTC', 'ETH/USD', 'ETH/BTC', 'ETC/BTC', 'ETC/USD', 'RRT/USD', 'RRT/BTC', 'ZEC/USD', 'ZEC/BTC', 'XMR/USD', 'XMR/BTC', 'DASH/USD', 'DASH/BTC', 'BTC/EUR', 'BTC/JPY', 'XRP/USD', 'XRP/BTC', 'IOTA/USD', 'IOTA/BTC', 'IOTA/ETH', 'EOS/USD', 'EOS/BTC', 'EOS/ETH', 'SAN/USD', 'SAN/BTC', 'SAN/ETH', 'OMG/USD', 'OMG/BTC', 'OMG/ETH', 'NEO/USD', 'NEO/BTC', 'NEO/ETH', 'ETP/USD', 'ETP/BTC', 'ETP/ETH', 'QTUM/USD', 'QTUM/BTC', 'QTUM/ETH', 'AVT/USD', 'AVT/BTC', 'AVT/ETH', 'EDO/USD', 'EDO/BTC', 'EDO/ETH', 'BTG/USD', 'BTG/BTC', 'DATA/USD', 'DATA/BTC', 'DATA/ETH', 'QASH/USD', 'QASH/BTC', 'QASH/ETH', 'YOYOW/USD', 'YOYOW/ETH', 'GNT/USD', 'GNT/BTC', 'GNT/ETH', 'SNT/USD', 'SNT/BTC', 'SNT/ETH', 'IOTA/EUR', 'BAT/USD', 'BAT/BTC', 'BAT/ETH', 'MANA/USD', 'MANA/BTC', 'MANA/ETH', 'FUN/USD', 'FUN/BTC', 'FUN/ETH', 'ZRX/USD', 'ZRX/BTC', 'ZRX/ETH', 'TNB/USD', 'TNB/BTC', 'TNB/ETH', 'SPANK/USD', 'SPANK/BTC', 'SPANK/ETH', 'TRX/USD', 'TRX/BTC', 'TRX/ETH', 'RCN/USD', 'RCN/BTC', 'RCN/ETH', 'RLC/USD', 'RLC/BTC', 'RLC/ETH', 'AID/USD', 'SNGLS/USD', 'SNGLS/BTC', 'SNGLS/ETH', 'REP/USD', 'REP/BTC', 'REP/ETH', 'ELF/USD', 'ELF/BTC', 'NEC/USD', 'NEC/BTC', 'NEC/ETH', 'BTC/GBP', 'ETH/EUR', 'ETH/JPY', 'ETH/GBP', 'NEO/EUR', 'NEO/JPY', 'NEO/GBP', 'EOS/EUR', 'EOS/JPY', 'EOS/GBP', 'IOTA/JPY', 'IOTA/GBP', 'IOST/USD', 'IOST/BTC', 'IOST/ETH', 'AION/USD', 'AION/BTC', 'AION/ETH', 'REQ/USD', 'REQ/BTC', 'REQ/ETH', 'RDN/USD', 'RDN/BTC', 'RDN/ETH', 'LRC/USD', 'LRC/BTC', 'LRC/ETH', 'WAXP/USD', 'WAXP/BTC', 'WAXP/ETH', 'DAI/USD', 'DAI/BTC', 'DAI/ETH', 'AGI/USD', 'AGI/BTC', 'AGI/ETH', 'BFT/USD', 'BFT/BTC', 'MTN/USD', 'MTN/ETH', 'ODE/USD', 'ODE/BTC', 'ODE/ETH', 'ANT/USD', 'ANT/BTC', 'ANT/ETH', 'DTH/USD', 'DTH/BTC', 'DTH/ETH', 'MITH/USD', 'MITH/BTC', 'STORJ/USD', 'STORJ/BTC', 'STORJ/ETH', 'XLM/USD', 'XLM/EUR', 'XLM/JPY', 'XLM/GBP', 'XLM/BTC', 'XLM/ETH', 'XVG/USD', 'XVG/EUR', 'XVG/GBP', 'XVG/BTC', 'MKR/USD', 'MKR/BTC', 'MKR/ETH', 'KNC/USD', 'KNC/BTC', 'KNC/ETH', 'POA/USD', 'POA/ETH', 'EVT/USD', 'LYM/USD', 'UTK/USD', 'UTK/BTC', 'UTK/ETH', 'VEE/USD', 'VEE/ETH', 'DADI/USD', 'DADI/ETH', 'ORS Group/USD', 'ORS Group/ETH', 'AUC/USD', 'AUC/BTC', 'POLY/USD', 'POLY/BTC', 'POLY/ETH', 'FSN/USD', 'FSN/BTC', 'CBT/USD', 'CBT/ETH', 'ZCN/USD', 'ZCN/BTC', 'ZCN/ETH', 'SEN/USD', 'SEN/ETH', 'NCASH/USD', 'CND/USD', 'CND/BTC', 'CND/ETH', 'CTXC/USD', 'CTXC/BTC', 'PAI/USD', 'PAI/BTC', 'SEER/USD', 'SEER/BTC', 'ESS/USD', 'ESS/ETH', 'ATMI/USD', 'Hydro Protocol/USD', 'Hydro Protocol/ETH', 'DTA/USD', 'DTA/BTC', 'DTA/ETH', 'IQ/USD', 'IQ/EOS', 'WPR/USD', 'WPR/BTC', 'WPR/ETH', 'ZIL/USD', 'ZIL/BTC', 'ZIL/ETH', 'BNT/USD', 'BNT/BTC', 'BNT/ETH', 'ABYSS/USD', 'ABYSS/ETH', 'XRA/USD', 'XRA/ETH', 'MAN/USD', 'NIO/USD', 'NIO/ETH', 'DGX/USD', 'DGX/ETH', 'VET/USD', 'VET/BTC', 'VET/ETH', 'UTNP/USD', 'TKN/USD', 'TKN/ETH', 'GOT/USD', 'GOT/EUR', 'GOT/ETH', 'XTZ/USD', 'XTZ/BTC', 'CNN/USD', 'CNN/ETH', 'BOX/USD', 'TRX/EUR', 'TRX/GBP', 'TRX/JPY', 'MGO/USD', 'MGO/ETH', 'RTE/USD', 'RTE/ETH', 'YGG/USD', 'MLN/USD', 'MLN/ETH', 'WTC/USD', 'WTC/ETH', 'CSX/USD', 'OMN/USD', 'OMN/BTC', 'INT/USD', 'INT/ETH', 'DRN/USD', 'DRN/ETH', 'PNK/USD', 'PNK/ETH', 'DGB/USD', 'DGB/BTC', 'BSV/USD', 'BSV/BTC', 'BCH/USD', 'BCH/BTC', 'WLO/USD', 'WLO/XLM', 'VLD/USD', 'ENJ/USD', 'ENJ/ETH', 'ONL/USD', 'RBT/USD', 'RBT/BTC', 'USDT/USD', 'EUT/EUR', 'EUT/USD', 'GUSD/USD', 'USDC/USD', 'TUSD/USD', 'PAX/USD', 'RIF/USD', 'RIF/BTC', 'PAS/USD', 'VSYS/USD', 'VSYS/BTC', 'MKR/DAI', 'BTT/USD', 'BTT/BTC', 'BTC/USDT', 'ETH/USDT', 'CLO/USD', 'CLO/BTC', 'IMP/USD', 'IMP/ETH', 'LTC/USDT', 'EOS/USDT', 'BCH/USDT', 'SCR/USD', 'GNO/USD', 'GNO/ETH', 'GEN/USD', 'ATOM/USD', 'ATOM/BTC', 'ATOM/ETH', 'WBT/USD', 'XCHF/USD', 'EUS/USD', 'WBT/ETH', 'XCHF/ETH', 'LEO/USD', 'LEO/BTC', 'LEO/USDT', 'LEO/EOS', 'LEO/ETH', 'AST/USD', 'AST/ETH', 'FOA/USD', 'UFR/USD', 'UFR/ETH', 'ZB/USD', 'ZB/USDT', 'OKB/USD', 'USK/USD', 'GTX/USD', 'KAN/USD', 'OKB/USDT', 'OKB/BTC', 'USK/USDT', 'USK/ETH', 'USK/BTC', 'USK/EOS', 'GTX/USDT', 'KAN/USDT', 'AMPL/USD', 'ALGO/USD', 'ALGO/BTC', 'ALGO/USDT', 'BTC/XCHF', 'SWM/USD', 'SWM/ETH', 'TRI/USD', 'LOO/USD', 'AMPL/USDT', 'DUSK/USD', 'DUSK/BTC', 'UOS/USD', 'UOS/BTC', 'RRB/USD', 'RRB/USDT', 'DTX/USD', 'DTX/USDT', 'AMPL/BTC', 'FTT/USD', 'FTT/USDT', 'PAX/USDT', 'USDC/USDT', 'TUSD/USDT', 'BTC/CNHT', 'USDT/CNHT', 'CNH/CNHT', 'CHZ/USD', 'CHZ/USDT', 'XAUT/USD', 'XAUT/BTC', 'XAUT/USDT', 'BTCF0/USTF0', 'ETHF0/USTF0', 'XAUTF0/USTF0']
```

I can obviously write logic to handle this situation internally for us, but it would be nice if these symbols were normalized like the rest.

**Comment by kroitor** [comment]
1. Bitfinex has renamed BCH multiple times:

- https://github.com/ccxt/ccxt/issues/5833
- https://github.com/ccxt/ccxt/issues/4989
- https://github.com/ccxt/ccxt/issues/4710
- https://github.com/ccxt/ccxt/search?q=bitfinex+bch&type=Issues

2. We can only parse symbols that are returned from the exchanges' markets endpoints. Unfortunately, you won't find BCHBTC here in the data, returned from Bitfinex v2 itself:

- https://api.bitfinex.com/v1/symbols

The problem is: bitfinex now calls it "BSVBTC" and there's no way to tell that "BCH" has ever existed with Bitfinex, using just the info from Bitfinex itself. So, in this case you have **very old** BCH trades, which CCXT is unable to unmap to unified symbols, because Bitfinex does not provide the information on that symbol (at all).

No BCH here: https://api.bitfinex.com/v1/symbols nor here https://api.bitfinex.com/v1/symbols_details

Let me know if the above does not answer the question.

**Comment by ethanbrown3** [comment]
I'm having the same issue with Bitfinex2 but it looks like to me that BCH/BTC is in the symbols endpoint. 
```
...
>>> ex = ccxt.bitfinex2(config)
>>> 'BCH/BTC' in ex.symbols
True
```
Yet when I fetch trades I have trades coming through with the symbol set to `tBCHBTC` rather then `BCH/BTC`

**Comment by ethanbrown3** [comment]
Here's the full trade dictionary just for kicks
```{
    "amount": 5,
    "cost": 0.6499,
    "datetime": "2017-08-18T14:00:32.000Z",
    "fee": {
        "cost": 5.559,
        "currency": "USD"
    },
    "id": "55075767",
    "info": [
        55075767,
        "tBCHBTC",
        1503064832000,
        3409788525,
        5,
        0.12998,
        null,
        null,
        -1,
        -5.559,
        "USD"
    ],
    "order": "3409788525",
    "price": 0.12998,
    "side": "buy",
    "symbol": "tBCHBTC",
    "takerOrMaker": "taker",
    "timestamp": 1503064832000,
    "type": null
}```

**Comment by kroitor** [comment]
> I'm having the same issue with Bitfinex2 but it looks like to me that BCH/BTC is in the symbols endpoint.

Nope. `exchange.symbols` is not an endpoint. It contains unified symbols after the markets are been loaded.

You are confusing the unified symbols from the unified exchange property here:

```
exchange.symbols
```

with the exchange-specific market ids from the exchange endpoint here:

- https://api.bitfinex.com/v1/symbols
- https://api.bitfinex.com/v1/symbols_details

The difference between the unified symbols and the market ids is explained in the Manual:

- https://github.com/ccxt/ccxt/wiki/Manual#symbols-and-market-ids

Here's an example that might shed the light for you (compare the symbol to id):

```
mbp:ccxt igorkroitor$ node examples/js/symbols bitfinex
Instantiating bitfinex exchange

Symbols:

tierBased | percentage | taker | maker | tiers           | id           | symbol             | base           | quote | baseId | quoteId | active
-------------------------------------------------------------------------------------------------------------------------------------------------
...
true      | true       | 0.002 | 0.001 | [object Object] | BABBTC       | BCH/BTC            | BCH            | BTC   | BAB    | BTC     | true
true      | true       | 0.002 | 0.001 | [object Object] | BABUSD       | BCH/USD            | BCH            | USD   | BAB    | USD     | true
true      | true       | 0.002 | 0.001 | [object Object] | BABUST       | BCH/USDT           | BCH            | USDT  | BAB    | UST     | true
...
```

So what you see is the lack of info from the exchange endpoint - bitfinex does not provide any info on the exchange specific market id tBCHBTC from any of the endpoints, therefore the lib does not recognize it as a unified market.

**Comment by ethanbrown3** [comment]
@kroiter Gotcha! Just wanted to make sure. I wasn't quite sure where `exchange.symbols` got it's data from. Thanks for your quick response. I'll implement my own workaround for my use case. Thanks!

**Comment by kroitor** [comment]
@ethanbrown3 @bobbytaxbit i am going to add a workaround that will recognize tBCHBTC with bitfinex2 properly within a few minutes.

**Comment by kroitor** [comment]
This should be fixed in CCXT 1.24.62+ it will arrive in 15 minutes when the Travis CI autobuild is finished. Let us know if you have further difficulties with it after upgrading.

**Comment by npomfret** [comment]
I've been wondering how the bitfinex reporting website does this, because it doesn't show the confusion.  Looks like the just map it

https://github.com/bitfinexcom/bfx-report-ui/blob/master/src/state/symbols/utils/mapping.js

Can't exactly see how, but the comments tell the story.
